### ISSUE TITLE
git pull/push connection error

### ISSUE THREAD

**Issue by 31337-4554551n** [flathub/com.jetpackduba.Gitnuro#6]
Running the app as a flatpak-and giving it flatseal permissions to access the whole drive-having configured git-config (and it working with command line git and vscodium flatpak) gitnuro gives connection issues when trying to pull/push.

It originally gave an error saying that included the URL of the git repo, but after the git-config settings in .gitconfig were modified with a second endry to point to "/run..." not just /usr the error changed to just connection error authentication dropped.

To Reproduce
Install gitnuro flatpak.
Configure git-config
ensure it works with git pull
Ensure it works with other flatpaks
ensure whole filesystem permissions for git nuro
Connection issues on pull/push

It to connect to the git repo and pull/push

    OS: pop is
    Version 22.04
    
When running `flatpak run com.jetpackduba.Gitnuro` and clicking pull:
```
java.lang.IllegalStateException: Cancelled authentication
	at com.jetpackduba.gitnuro.credentials.HttpCredentialsProvider.handleExternalCredentialHelper(HttpCredentialsProvider.kt:154)
	at com.jetpackduba.gitnuro.credentials.HttpCredentialsProvider.get(HttpCredentialsProvider.kt:57)
	at org.eclipse.jgit.transport.HttpAuthMethod.authorize(HttpAuthMethod.java:205)
	at org.eclipse.jgit.transport.TransportHttp.connect(TransportHttp.java:683)
	at org.eclipse.jgit.transport.TransportHttp.openFetch(TransportHttp.java:465)
	at org.eclipse.jgit.transport.FetchProcess.executeImp(FetchProcess.java:151)
	at org.eclipse.jgit.transport.FetchProcess.execute(FetchProcess.java:103)
	at org.eclipse.jgit.transport.Transport.fetch(Transport.java:1405)
	at org.eclipse.jgit.api.FetchCommand.call(FetchCommand.java:233)
	at org.eclipse.jgit.api.PullCommand.call(PullCommand.java:266)
	at com.jetpackduba.gitnuro.git.remote_operations.PullBranchUseCase$invoke$2.invokeSuspend(PullBranchUseCase.kt:19)
	at com.jetpackduba.gitnuro.git.remote_operations.PullBranchUseCase$invoke$2.invoke(PullBranchUseCase.kt)
	at com.jetpackduba.gitnuro.git.remote_operations.PullBranchUseCase$invoke$2.invoke(PullBranchUseCase.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:89)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:166)
	at kotlinx.coroutines.BuildersKt.withContext(Unknown Source)
	at com.jetpackduba.gitnuro.git.remote_operations.PullBranchUseCase.invoke(PullBranchUseCase.kt:13)
	at com.jetpackduba.gitnuro.viewmodels.MenuViewModel$pull$1.invokeSuspend(MenuViewModel.kt:26)
	at com.jetpackduba.gitnuro.viewmodels.MenuViewModel$pull$1.invoke(MenuViewModel.kt)
	at com.jetpackduba.gitnuro.viewmodels.MenuViewModel$pull$1.invoke(MenuViewModel.kt)
	at com.jetpackduba.gitnuro.git.TabState$safeProcessing$1$2.invokeSuspend(TabState.kt:73)
	at com.jetpackduba.gitnuro.git.TabState$safeProcessing$1$2.invoke(TabState.kt)
	at com.jetpackduba.gitnuro.git.TabState$safeProcessing$1$2.invoke(TabState.kt)
	at com.jetpackduba.gitnuro.extensions.StateManagementUtilsKt.delayedStateChange(StateManagementUtils.kt:21)
	at com.jetpackduba.gitnuro.git.TabState$safeProcessing$1.invokeSuspend(TabState.kt:67)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:749)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
LastError flow: Error(date=1680232859541, exception=java.lang.IllegalStateException: Cancelled authentication, message=Cancelled authentication)
```

**Comment by JetpackDuba** [comment]
Are you using a credentials helper? Like https://github.com/git-ecosystem/git-credential-manager
It seems that it fails to run it.

**Comment by 31337-4554551n** [comment]
> Are you using a credentials helper? Like https://github.com/git-ecosystem/git-credential-manager It seems that it fails to run it.

I am, yes, that's what I am using.
It works for my regular git, and also for vscode

I tried using it normally, and also putting /run/usr in front of the line in .gitconfig as well after seeing it in a support forum for vscode. I think that's meant to help with flatpaks. I also tried giving gitnuro full access to the root folder in flatseal, just in case it couldn't access that program (I am unsure why that wouldn't be a default if it requires it though).
But yes. I am using the credential manager.

**Comment by JetpackDuba** [comment]
Hey!

I've found the issue. It's a mix between Gitnuro not knowing it's running on flatpak and running thing in the wrong way. The flatpak package should also specify a env var and the DBus socket (I think it's needed, I'll do further testing). As it requires changes in both sides, I can't do it as quickly as I'd like, but I'll reopen the issue in the Gitnuro repo and include it in 1.2.0 (which should be released in 2-3 weeks).

I'll also keep the issue open here as it's something that has to be changed in both sides.

Thank you for your report!

**Comment by 31337-4554551n** [comment]
That's great, thank you! 

So once that's updated will I need to modify flatseal permissions or add the /run to the start of the path to manager in gitconfig?
Or it should work out of the box?

**Comment by JetpackDuba** [comment]
AFAIK, it should work out of the box.

**Comment by miroirdelame** [comment]
Same behavior for me. I switched to the Java version for the meantime.

**Comment by 31337-4554551n** [comment]
Just checking in, curious if this is still looking likely in a week or so?

**Comment by JetpackDuba** [comment]
Hey! 

Yeah, towards the end of the next week but yes (so I can ask a friend to help me test the MacOS build during the weekend). I've got a fix for this issue ready and I'm currently working on finishing the remaining issues for the release.

**Comment by JetpackDuba** [comment]
The release is almost ready (All the issues have been closed in the Gitnuro repo) but I've got some issues with the build, I expect having it ready between tomorrow and wednesday.

Sorry for the delay.

**Comment by JetpackDuba** [comment]
The release for the JAR is already out but the Flatpak not yet due to this blocking PR https://github.com/flathub/flatpak-builder-lint/pull/138

I need specific permission to launch an external process outside the flatpak sandboxing and that requires requesting such permissiong before adding/updating the Flathub app.

Once this is accepted, I can merge this https://github.com/flathub/com.jetpackduba.Gitnuro/pull/7

**Comment by 31337-4554551n** [comment]
I'm very much a novice to this whole git process.
Could you please explain who you're getting permission from, if you yourself are the dev?

**Comment by JetpackDuba** [comment]
Sure! 

The Flatpak version of Gitnuro is published on Flathub. The process is automated, so I just have to update the Git repository by creating a pull request. The thing is that every release has first to be built by a bot. This bot verifies the new release file and also does some verifications to ensure the users' security (because flatpak apps are sandboxed by default).

In this new release, I'm adding functionality that runs outside the sandboxing (to run the credentials manager and the option to start the system terminal), therefore I need a special permission that has to be accepted by the Flathub admins. The bot will reject the update until Gitnuro is added as an exception to use such feature.

If before this weekend I don't get an answer, I'll try to ping the admins again, because I want to release v1.2.1.

If you could give a try to the 1.2.0 branch and let me know if it works for you, it would be great :smile:



These are the steps in case you want to try it:

```
git clone https://github.com/flathub/com.jetpackduba.Gitnuro.git
cd com.jetpackduba.Gitnuro
git checkout 1.2.0
flatpak-builder build-dir com.jetpackduba.Gitnuro.yml --force-clean
flatpak-builder --user --install --force-clean build-dir com.jetpackduba.Gitnuro.yml
```

You may have to install flatpak-builder if you haven't got it: `sudo apt install flatpak-builder`

**Comment by JetpackDuba** [comment]
Hello again,

I finally got the permission approved and I've directly pushed version 1.2.1 (just released today). Credentials manager works out of the box without having to do anything special. It'll be available in Flathub in a few hours (I think about ~3h).

Feel free to reopen this issue if the problem persists :smiley:

**Comment by 31337-4554551n** [comment]
Hi,

Thank you so much. Just updated, opened it, opened a repo, tried to pull and got:

Error
https://[repo]: Cannot open git-upload-pack

Tried again, and got
Error
Cancelled Authentication

I am unable to re-open the issue

**Comment by JetpackDuba** [comment]
Does it work without Credenrials Manager? To discard possible issues.

You should also find logs in $HOME/.gitnuro

**Comment by 31337-4554551n** [comment]
2023-05-15 14:45:16 ERROR slf4j:16 - TabState - Cancelled authentication
java.lang.IllegalStateException: Cancelled authentication
	at com.jetpackduba.gitnuro.credentials.HttpCredentialsProvider.handleExternalCredentialHelper(HttpCredentialsProvider.kt:158)
	at com.jetpackduba.gitnuro.credentials.HttpCredentialsProvider.get(HttpCredentialsProvider.kt:62)
	at org.eclipse.jgit.transport.HttpAuthMethod.authorize(HttpAuthMethod.java:205)
	at org.eclipse.jgit.transport.TransportHttp.connect(TransportHttp.java:683)
	at org.eclipse.jgit.transport.TransportHttp.openFetch(TransportHttp.java:465)
	at org.eclipse.jgit.transport.FetchProcess.executeImp(FetchProcess.java:153)
	at org.eclipse.jgit.transport.FetchProcess.execute(FetchProcess.java:105)
	at org.eclipse.jgit.transport.Transport.fetch(Transport.java:1462)
	at org.eclipse.jgit.api.FetchCommand.call(FetchCommand.java:238)
	at org.eclipse.jgit.api.PullCommand.call(PullCommand.java:266)
	at com.jetpackduba.gitnuro.git.remote_operations.PullBranchUseCase$invoke$2.invokeSuspend(PullBranchUseCase.kt:27)
	at com.jetpackduba.gitnuro.git.remote_operations.PullBranchUseCase$invoke$2.invoke(PullBranchUseCase.kt)
	at com.jetpackduba.gitnuro.git.remote_operations.PullBranchUseCase$invoke$2.invoke(PullBranchUseCase.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:89)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:169)
	at kotlinx.coroutines.BuildersKt.withContext(Unknown Source)
	at com.jetpackduba.gitnuro.git.remote_operations.PullBranchUseCase.invoke(PullBranchUseCase.kt:15)
	at com.jetpackduba.gitnuro.viewmodels.MenuViewModel$pull$1.invokeSuspend(MenuViewModel.kt:35)
	at com.jetpackduba.gitnuro.viewmodels.MenuViewModel$pull$1.invoke(MenuViewModel.kt)
	at com.jetpackduba.gitnuro.viewmodels.MenuViewModel$pull$1.invoke(MenuViewModel.kt)
	at com.jetpackduba.gitnuro.git.TabState$safeProcessing$job$1$2.invokeSuspend(TabState.kt:131)
	at com.jetpackduba.gitnuro.git.TabState$safeProcessing$job$1$2.invoke(TabState.kt)
	at com.jetpackduba.gitnuro.git.TabState$safeProcessing$job$1$2.invoke(TabState.kt)
	at com.jetpackduba.gitnuro.extensions.StateManagementUtilsKt.delayedStateChange(StateManagementUtils.kt:21)
	at com.jetpackduba.gitnuro.git.TabState$safeProcessing$job$1.invokeSuspend(TabState.kt:119)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
	
	
	Now also getting
	Error
	https://[repo]: authentication not supported
	
	2023-05-15 15:43:19 ERROR slf4j:16 - TabState - [repo]: authentication not supported
org.eclipse.jgit.api.errors.TransportException: [repo]: authentication not supported
	at org.eclipse.jgit.api.FetchCommand.call(FetchCommand.java:249)
	at org.eclipse.jgit.api.PullCommand.call(PullCommand.java:266)
	at com.jetpackduba.gitnuro.git.remote_operations.PullBranchUseCase$invoke$2.invokeSuspend(PullBranchUseCase.kt:27)
	at com.jetpackduba.gitnuro.git.remote_operations.PullBranchUseCase$invoke$2.invoke(PullBranchUseCase.kt)
	at com.jetpackduba.gitnuro.git.remote_operations.PullBranchUseCase$invoke$2.invoke(PullBranchUseCase.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:89)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:169)
	at kotlinx.coroutines.BuildersKt.withContext(Unknown Source)
	at com.jetpackduba.gitnuro.git.remote_operations.PullBranchUseCase.invoke(PullBranchUseCase.kt:15)
	at com.jetpackduba.gitnuro.viewmodels.MenuViewModel$pull$1.invokeSuspend(MenuViewModel.kt:35)
	at com.jetpackduba.gitnuro.viewmodels.MenuViewModel$pull$1.invoke(MenuViewModel.kt)
	at com.jetpackduba.gitnuro.viewmodels.MenuViewModel$pull$1.invoke(MenuViewModel.kt)
	at com.jetpackduba.gitnuro.git.TabState$safeProcessing$job$1$2.invokeSuspend(TabState.kt:131)
	at com.jetpackduba.gitnuro.git.TabState$safeProcessing$job$1$2.invoke(TabState.kt)
	at com.jetpackduba.gitnuro.git.TabState$safeProcessing$job$1$2.invoke(TabState.kt)
	at com.jetpackduba.gitnuro.extensions.StateManagementUtilsKt.delayedStateChange(StateManagementUtils.kt:21)
	at com.jetpackduba.gitnuro.git.TabState$safeProcessing$job$1.invokeSuspend(TabState.kt:119)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
Caused by: org.eclipse.jgit.errors.TransportException: [repo]: authentication not supported
	at org.eclipse.jgit.transport.TransportHttp.connect(TransportHttp.java:674)
	at org.eclipse.jgit.transport.TransportHttp.openFetch(TransportHttp.java:465)
	at org.eclipse.jgit.transport.FetchProcess.executeImp(FetchProcess.java:153)
	at org.eclipse.jgit.transport.FetchProcess.execute(FetchProcess.java:105)
	at org.eclipse.jgit.transport.Transport.fetch(Transport.java:1462)
	at org.eclipse.jgit.api.FetchCommand.call(FetchCommand.java:238)
	... 24 more

**Comment by JetpackDuba** [comment]
The logs are quite odd, I'm not really sure what's the issue.

1. Can you test with the JAR instead of flatpak? To discard possible issues with the sandboxing. 
1. Can you also share your credentials config in .gitconfig?

For example I'm using the following config: 
```
[credential]
	helper = /usr/bin/git-credential-manager
	credentialStore = cache
[credential "https://dev.azure.com"]
	useHttpPath = true
```

**Edit**

Before the errors, the logs should show something similar to this:
`2023-05-15 10:57:37 INFO  slf4j:8 - ShellManager - runCommandProcess: /usr/bin/flatpak-spawn --host --env=TERM=xterm-256color /usr/bin/git-credential-manager get`

**Comment by 31337-4554551n** [comment]
My equivalent log line I believe is

2023-05-15 15:43:19 INFO  slf4j:8 - ShellManager - runCommandProcess: /usr/bin/flatpak-spawn --host --env=TERM=xterm-256color /run/host/usr/local/bin/git-credential-manager get

.gitconfig:

[credential]
	helper = /usr/local/bin/git-credential-manager
	credentialStore = secretservice
        helper = /run/host/usr/local/bin/git-credential-manager
	credentialStore = secretservice
[credential "https://dev.azure.com"]
	useHttpPath = true
[credential "[repo]"]
	provider = github

**Comment by JetpackDuba** [comment]
You have the helper specified twice, being the second one invalid, since you don't need to prefix it with `/run/host`.

Try with this:

```
[credential]
    helper = /usr/local/bin/git-credential-manager
    credentialStore = secretservice
[credential "https://dev.azure.com"]
    useHttpPath = true
[credential "[repo]"]
    provider = github
```

**Comment by 31337-4554551n** [comment]
That worked, thank you.
Strange that since you're saying the second one being invalid was the issue, when the first one is the one that needed to stay. But either way, it's working now, thank you so much!!

**Comment by JetpackDuba** [comment]
Exactly, the second being invalid means that it had to go and the first had to stay :smile: 

Whatever, glad to help! Let me know if you face any other issues.

**Comment by 31337-4554551n** [comment]
Sorry, what I meant was, if it was invalid, and the first in the list was valid, why did the invalid one, further down the list, break the connection instead of just being ignored?

**Comment by JetpackDuba** [comment]
Ah! Because it takes the latest matching option (in this case, the latest "helper") and it's used afterwards, it doesn't know if it's valid or not until it has to be used, at load time any value is valid.

In other words, it loads all the configs using a key-value based structure that can only have a single value for each key.

**Comment by 31337-4554551n** [comment]
Ah, LATEST matching, not first. And that one is broken, etc etc.
Ok, that makes sense. thank you again for all your help!

**Comment by 31337-4554551n** [comment]
So it turns out I actually needed

helper = /run/host/usr/local/bin/git-credential-manager
credentialStore = secretservice

To make visualstudio codium function.

But I need
helper = /usr/local/bin/git-credential-manager
credentialStore = secretservice

For ginuro.

Removing the second one means that vs codium can no longer connect to git.
Is there a way to make them both work?

**Comment by 31337-4554551n** [comment]
I know vscode isn't your problem in any way. I am just wondering if there's a way to make different flatpaks play nice when it comes to sharing the git credentials manager

**Comment by JetpackDuba** [comment]
Sorry I'm sick, I will try to investigate it when I feel better

**Comment by 31337-4554551n** [comment]
Take care of yourself, hope you get well soon!

**Comment by JetpackDuba** [comment]
Thank you :smiley: 

Give this a try:

Create a softlink of the credentials manager to a path in your PC matching the path in the Flatpak sandboxing. Based on your config, I should be something like this:

```
sudo mkdir -p /run/host/usr/local/bin/
sudo ln -s  /usr/local/bin/git-credential-manager /run/host/usr/local/bin/
```

Update `.gitconfig` to use `/run/host/usr/local/bin/git-credential-manager`

**Comment by 31337-4554551n** [comment]
Oh you're an absolute legend!
I'm surprised /run/host/usr/local/bin/ didn't already exist, given that config worked earlier for visual studio, but following your instructions works perfectly!
I'm still not quite understanding what that /run/host is, and why it doesn't universally work for all flatpaks, and some need it, others specifically don't. It's a bit of a magic config atm, could you please let me know the logic behind it all?

**Comment by JetpackDuba** [comment]
Flatpak has a sandboxing to limit the access  of the applications to the file system. As an example, the previous version of Gitnuro only had access to the $HOME directory, but not anymore because I needed access to rest of the filesystem to launch external terminal emulators and credentials manager itself. 

To do all this sandboxing, flatpak has some kind of virtual filesystem (not sure if it's the right term), which also contains a specific version of the app dependencies installed in this file system, for Gitnuro's case that would be a java installation that is only accessible by Gitnuro (well, the same installation would be accessible for other apps that use the same Java dependency to avoid wasting space). To provide access to some parts of the physical filesystem, Flatpak mounts  such parts in `run/host`. 

The latest release of Gitnuro, rather than relying on this aproach that requires adapting `.gitconfig` to set the workaround path, uses an additional permission (the one that required being accepted by Flathub admins a few weeks ago) that allows running commands in the user's filesystem.

In other words, VSCode (Flatpak version) is running in a sandboxed environment where Flatpak provides access to the user's filesystem in a very specific way (using `run/host`), while Gitnuro bypasses these sandboxing restrictions.

**Comment by 31337-4554551n** [comment]
I see! So vscode doesn't have the same permission, and so mounts that part into run/host
Requiring messing with .gitconfig

I assume gitnuro could have done that, but didn't want users to then have to change their .gitconfig?

**Comment by JetpackDuba** [comment]
Yeah, exactly.

**Comment by 31337-4554551n** [comment]
Continuing my own stupidity:

I'm currently setting
sudo mkdir -p /run/host/usr/local/bin/
sudo ln -s  /usr/local/bin/git-credential-manager /run/host/usr/local/bin/
In a startup script, because it resets regularly.

I'm sure I'm not the only one using multiple flatpaks/apps with different git credential managers.

How should I be handling something like this?

And I know you've put in so much more than you ever needed to into this, so thank you so much once again.

**Comment by JetpackDuba** [comment]
> Continuing my own stupidity:
> 
> I'm currently setting sudo mkdir -p /run/host/usr/local/bin/ sudo ln -s /usr/local/bin/git-credential-manager /run/host/usr/local/bin/ In a startup script, because it resets regularly.
> 

Ah, sorry! I completly forgot to mention that /run is wiped at boot time, my bad.


> I'm sure I'm not the only one using multiple flatpaks/apps with different git credential managers.

Perhaps you are :smile: Or at least the only mention I've seen about it. Is there any reason to use vscode integrated Git rather than Gitnuro?

> How should I be handling something like this?

I guess that your approach is the correct one. I'd run the script in a systemd service or even a crontab (root crontab) that runs it at boot.

> And I know you've put in so much more than you ever needed to into this, so thank you so much once again.

No worries!

**Comment by 31337-4554551n** [comment]
> Is there any reason to use vscode integrated Git rather than Gitnuro

I use both.
Just for my workflow, I need to go through and look at/troubleshoot configs on master mostly.
So I open vscode, and click the little sync button down the bottom to make sure I'm looking at the latest.
Then when I actually need to do changes, look at history, see who did what, that's when I use gitnuro.

> I'd run the script in a systemd service or even a crontab (root crontab) that runs it at boot

Guess I better learn how to do that. Seems like a useful thing to know, especially about creating my own systemd service.
Though I'm amazed I'm the only person in the world doing this.
Maybe I'll open a support case with vscode to get them to do what you did haha. Make life easier for all :P
